Libraries/LibJS/AST.cpp

```diff
@@ -280,6 +282,139 @@ Value ForStatement::execute(Interpreter& interpreter) const
     return last_value;
 }
 
+static FlyString variable_from_for_declaration(Interpreter& interpreter, NonnullRefPtr<ASTNode> node, RefPtr<BlockStatement> wrapper)
+{
+    FlyString variable_name;
+    if (node->is_variable_declaration()) {
+        auto* variable_declaration = static_cast<const VariableDeclaration*>(node.ptr());
+        ASSERT(!variable_declaration->declarations().is_empty());
+        if (variable_declaration->declaration_kind() != DeclarationKind::Var) {
+            wrapper = create_ast_node<BlockStatement>();
+            interpreter.enter_scope(*wrapper, {}, ScopeType::Block);
+        }
+        variable_declaration->execute(interpreter);
+        variable_name = variable_declaration->declarations().first().id().string();
+    } else if (node->is_identifier()) {
+        variable_name = static_cast<const Identifier&>(*node).string();
+    } else {
+        ASSERT_NOT_REACHED();
+    }
+    return variable_name;
+}
+
+Value ForInStatement::execute(Interpreter& interpreter) const
+{
+    if (!m_lhs->is_variable_declaration() && !m_lhs->is_identifier()) {
+        // Apparently this is legal in some cases, e.g. for (a.b in c) {}
```

Comment:
Seems like it's taken as a assignment?
```js
for (a.b in c) stmt
```
->
```js
for (let _t in c) { a.b = _t; stmt; }
```

Reply:
Mystery solved, thanks!

Confirmed using an object setter function in a mature:tm: engine:

```js
var o = {
    _foo: undefined,
    get foo() {
        console.log("foo was accessed, its value is " + this._foo);
    },
    set foo(value) {
        this._foo = value;
        console.log("foo was set to " + value);
    }
}
for (o.foo in "hello") {
    o.foo; 
}
```

```
foo was set to 0
foo was accessed, its value is 0
foo was set to 1
foo was accessed, its value is 1
foo was set to 2
foo was accessed, its value is 2
foo was set to 3
foo was accessed, its value is 3
foo was set to 4
foo was accessed, its value is 4
```